Engines/Wine/Verbs/xact/script.js

```diff
@@ -11,50 +11,51 @@ include("engines.wine.plugins.regsvr32");
  * @returns {Wine} Wine object
  */
 Wine.prototype.xact = function () {
-    var that = this;
-
-    var extractFiles = function (progressBar, filesToExtract, destination, pattern, directory) {
-        var numberOfExtractedFiles = 0;
-        filesToExtract.forEach(function (cabFile) {
+    const extractFiles = (progressBar, filesToExtract, destination, pattern, directory) => {
+        // extract the cab files
+        filesToExtract.reduce((numberOfExtractedFiles, cabFile) => {
             print(tr("Extracting {0}...", cabFile));
+
             progressBar.setText(tr("Extracting {0}...", "Xact"));
-            progressBar.setProgressPercentage(numberOfExtractedFiles * 100 / filesToExtract.length);
+            progressBar.setProgressPercentage((numberOfExtractedFiles * 100) / filesToExtract.length);
 
             new CabExtract()
-                .archive(that.prefixDirectory() + "/drive_c/" + directory + cabFile)
+                .archive(this.prefixDirectory() + "/drive_c/" + directory + cabFile)
                 .to(destination)
                 .extract(["-L", "-F", pattern]);
 
-            numberOfExtractedFiles++;
-        });
+            return numberOfExtractedFiles + 1;
+        }, 0);
     };
 
     //This function executes regsvr32 on the dlls present in dllToRegsvr
-    var regsvr32Xact = function (progressBar, dllToRegsvr) {
-        var numberOfExtractedFiles = 0;
-        dllToRegsvr.forEach(function (dll) {
+    const regsvr32Xact = (progressBar, dllToRegsvr) => {
+        dllToRegsvr.reduce((numberOfExtractedFiles, dll) => {
             print(tr("Registering {0}...", dll));
+
             progressBar.setText(tr("Registering {0}...", "Xact"));
-            progressBar.setProgressPercentage(numberOfExtractedFiles * 100 / filesToExtract.length);
+            progressBar.setProgressPercentage((numberOfExtractedFiles * 100) / filesToExtract.length);
 
-            that.regsvr32().install(dll);
+            this.regsvr32().install(dll);
 
-            numberOfExtractedFiles++;
-        });
+            return numberOfExtractedFiles + 1;
+        }, 0);
     };
 
-    var setupFile = new Resource()
+    const setupFile = new Resource()
         .wizard(this.wizard())
-        .url("http://download.microsoft.com/download/8/4/A/84A35BF1-DAFE-4AE8-82AF-AD2AE20B6B14/directx_Jun2010_redist.exe")
+        .url(
+            "http://download.microsoft.com/download/8/4/A/84A35BF1-DAFE-4AE8-82AF-AD2AE20B6B14/directx_Jun2010_redist.exe"
+        )
         .checksum("f8f1217f666bf2f6863631a7d5e5fb3a8d1542df")
         .name("directx_Jun2010_redist.exe")
         .get();
 
-    var progressBar = this.wizard().progressBar(tr("Please wait..."));
+    const progressBar = this.wizard().progressBar(tr("Please wait..."));
     progressBar.setText(tr("Extracting {0}...", "Xact"));
-    progressBar.setProgressPercentage(0.);
+    progressBar.setProgressPercentage(0);
 
-    var filesToExtract = []
+    let filesToExtract = [];
```

Comment:
I don't like it that we override `filesToExtract` later on. I think it would be better to use multiple variables for the different arrays.